python/mage/node_classification/utils/extract_from_database.py

```diff
@@ -0,0 +1,262 @@
+import torch
+from torch_geometric.data import Data, HeteroData
+import mgp
+import numpy as np
+from tqdm import tqdm
+import random
+from collections import Counter
+import torch_geometric.transforms as T
+import typing
+
+def extract_from_database(
+    ctx: mgp.ProcCtx,
+    train_ratio: float,
+    features_name: str,
+    class_name: str,
+) -> typing.Tuple[HeteroData, str, typing.Dict, typing.Dict]:
+    
+    data = HeteroData()
+    
+
+    #################
+    # NODES
+    #################
+    nodes = list(iter(ctx.graph.vertices))
+    node_types = []
+    embedding_lengths = {}
+    observed_attribute = None
+    
+    for i in range(len(nodes)):
```

Comment:
Since index `i` isn't used anywhere else, you can use for each loop:

`for node in nodes:`

Reply:
Thanks, I learned small, but very important thing from this comment.